core/src/mender-artifact.c

```diff
@@ -1224,10 +1224,11 @@ artifact_read_data_prepare(mender_artifact_ctx_t *ctx, mender_artifact_download_
         mender_log_error("Unable to allocate memory");
         return MENDER_FAIL;
     }
-    for (char *ch = strstr(mdata_cache->checksum_fname, ".tar"); (NULL != ch) && (*ch != '\0'); ch++) {
+    char *ch = mdata_cache->checksum_fname;
+    while (NULL != (ch = strstr(ch, ".tar"))) {
         /* Don't worry! The call to strlen() on a static string should
          * be optimized out by the compiler */
-        *ch = ch[strlen(".tar")];
+        memmove(ch, ch + strlen(".tar"), strlen(ch + strlen(".tar")) + 1);
```

Comment:
This was the only way I could get the address sanitizer to pass, the simple pointer arithmetic kept throwing heap buffer overflow errors, and I couldn't find another way to work around it

Reply:
It looks good to me but I am looking forward to hear from our C experts too :monocle_face:

Reply:
I'm a bit hesitant to use sub-optimal code only to make a (buggy, right?) sanitizer happy. What exactly did it not like? And did it give any details?

Reply:
https://github.com/mendersoftware/mender-mcu/pull/125